language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandExecutor.java

```diff
@@ -180,6 +181,20 @@ private static void executeAddDocumentation(WorkspaceServiceContext context) {
             case UtilSymbolKeys.TRANSFORMER_KEYWORD_KEY:
                 documentationContent = CommandUtil.getTransformerDocumentation(bLangPackage, line);
                 break;
+            case UtilSymbolKeys.RESOURCE_KEYWORD_KEY:
+                CommandUtil.DocAttachmentInfo attachInfo = CommandUtil.getResourceDocumentation(bLangPackage, line);
+                if (attachInfo != null) {
+                    documentationContent = attachInfo.getDocAttachment();
+                    line = attachInfo.getReplaceStartFrom();
+                }
```

Comment:
Shall we move if scope to a granular method and reuse it below case as well?

Reply:
This particular logic is being refactored with the new feature implementation of "Adding documentation for the whole document". This change will be included with that.